src/grid/submission/SubmissionGridHeader.component.ts

```diff
@@ -1,30 +1,66 @@
-import { Component } from '@angular/core';
-import { Utils, Components } from 'formiojs';
-import { GridHeaderComponent } from '../GridHeaderComponent';
+import {Component, EventEmitter} from '@angular/core';
+import {Utils, Components, ExtendedComponentSchema} from 'formiojs';
+import {GridHeaderComponent} from '../GridHeaderComponent';
 import {FormioPromiseService} from '../../formio-promise.service';
+import {FormioForm} from '../../formio.common';
 
 @Component({
   templateUrl: './SubmissionGridHeader.component.html'
 })
 export class SubmissionGridHeaderComponent extends GridHeaderComponent {
-  load(formio: FormioPromiseService, query?: any) {
+
+  // Map structure where the key is the path and the value is the component
+  formComponents: Map<string, ExtendedComponentSchema>;
+
+  load(formio: FormioPromiseService, query?: any, columns?: Array<any>) {
     query = query || {};
-    return formio.loadForm({params: query}).then((form: any) => {
+    return formio.loadForm({params: query}).then((form: FormioForm) => {
       this.headers = [];
-      Utils.eachComponent(form.components, (component: any) => {
-        if (
-          component.input &&
-          (!component.hasOwnProperty('tableView') || component.tableView)
-        ) {
-          this.headers.push({
-            label: component.label,
-            key: 'data.' + component.key,
-            sort: '',
-            component: Components.create(component, null, null, true)
-          });
+      this.formComponents = new Map<string, ExtendedComponentSchema>();
+      this.setComponents(form.components);
+      columns ? columns.forEach(column => {
+        if (this.formComponents.has(column.path)) {
+          this.setHeader(column, this.formComponents.get(column.path));
+        } else {
+          this.setHeader(column);
         }
-      }, true);
+      }) : this.setComponentsHeaders(this.formComponents);
+
       return this.headers;
     });
   }
+
+  // Set header for both component and column
+  setHeader(column?: any, component?: ExtendedComponentSchema, sort?: EventEmitter<any>) {
+    const key = column ? column.path : `data.${component.key}`;
+    const label = column ? column.label : component.label;
+
+    this.headers.push({
```

Comment:
@vlad-shusterman also I see that you don't pass `renderCell` method (`getView` from your current code) here from column to header, so `header.getView` will always be empty, so line 29 from `SubmissionGridBody.component.ts ` will never be executed